src/app/content/highlights/components/Highlights.tsx

```diff
@@ -134,3 +206,145 @@ export const SectionHighlights = ({ location, highlights }: SectionHighlightsPro
+}: HighlightToggleEditProps) => <Styled.HighlightToggleEdit>
+  <Dropdown
+    toggle={<MenuToggle/>}
+    className='highlight-toggle-edit'
+  >
+    <Styled.HighlightToggleEditContent>
+      <ColorPicker
+        color={color}
+        onChange={onColorChange}
+        onRemove={() => onDelete()}
+      />
+      <DropdownList>
+        <DropdownItem
+          message='i18n:highlighting:dropdown:edit'
+          prefix={<EditIcon/>}
+          onClick={() => onEdit()}
+        />
+        <DropdownItem
+          message='i18n:highlighting:dropdown:delete'
+          prefix={<TrashAltIcon/>}
+          onClick={() => onDelete()}
+        />
+      </DropdownList>
+    </Styled.HighlightToggleEditContent>
+  </Dropdown>
+</Styled.HighlightToggleEdit>;
+
+interface HighlightAnnotationProps {
+  annotation: string;
+  isEditable: boolean;
+  onSave: (annotation: string) => void;
+  onCancel: () => void;
+}
+
+// tslint:disable-next-line:variable-name
+const HighlightAnnotation = (
+  { annotation, isEditable, onSave, onCancel }: HighlightAnnotationProps
+) => {
+  const [anno, setAnno] = React.useState(annotation);
+
+  const onKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      onCancel();
+      return;
+    } else if (e.key === 'Enter' && !e.shiftKey) {
+      onSave(anno);
+      return;
+    }
+  };
+
+  return <Styled.HighlightNote>
+    {isEditable
+      ? <FormattedMessage id='i18n:highlighting:card:placeholder'>
+        {(msg: string) => <textarea
+          value={anno}
+          placeholder={msg}
+          autoFocus
+          onKeyDown={onKeyDown}
+          onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => {
+            setAnno((e.target as { value: string }).value);
+          }}
+        />}
+      </FormattedMessage>
+      : <Styled.HighlightNote>
+        <span>
+          <FormattedMessage id='i18n:toolbar:highlights:popup:body:note:text'>
+            {(msg: Element | string) => msg}
+          </FormattedMessage>
+        </span>
+        {annotation}
+      </Styled.HighlightNote>
+      }
+    {isEditable && <Styled.HighlightEditButtons>
+      <FormattedMessage id='i18n:highlighting:button:save'>
+        {(msg: Element | string) => <Button
+          data-testid='save'
+          data-analytics-label='save'
+          size='medium'
+          variant='primary'
+          onClick={() => onSave(anno)}
+        >{msg}</Button>}
+      </FormattedMessage>
+      <FormattedMessage id='i18n:highlighting:button:cancel'>
+        {(msg: Element | string) => <Button
+          size='medium'
+          data-analytics-label='cancel'
+          data-testid='cancel'
+          onClick={onCancel}
+        >{msg}</Button>}
+      </FormattedMessage>
+    </Styled.HighlightEditButtons>}
+  </Styled.HighlightNote>;
+};
+
+interface HighlightDeleteWrapperProps {
+  onCancel: () => void;
+  onDelete: () => void;
+}
+
+// tslint:disable-next-line:variable-name
+const HighlightDeleteWrapper = ({
```

Comment:
i think you split these components into new files in a different PR, but you can go ahead and do it on this branch, it would be easier to read with them in different files